### PR TITLE
Include filter for image attachements

### LINKED ISSUE #1644: Create image media endpoint
Current endpoint provided by Onadata gets all media forms i.e videos, audio and images.
`curl -X GET https://api.ona.io/api/v1/media.json?page=1&page_size=4`

Requesting for an end point that gets images only  and allows pagination

### PR DESCRIPTION
**Pull request by WinnyTroy**
Fixes #1644

### CODE REVIEW

File: onadata/libs/filters.py [hunk 1]
```diff
@@ -399,6 +399,19 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class ImageAttachmentFilter(filters.BaseFilterBackend):
+
+    def filter_queryset(self, request, queryset, view):
+        image_set = ["image/png", "image/jpeg", "image/jpg"]
+        mimetype = request.query_params.get('type')
```
[the remark this answers is not in the document]
Comment (ukanga) [reply onadata/libs/filters.py]: I think the variable `mimetype` is misleading, perhaps `attachment_type`?

File: onadata/libs/filters.py [hunk 2]
```diff
@@ -399,6 +399,19 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class ImageAttachmentFilter(filters.BaseFilterBackend):
+
+    def filter_queryset(self, request, queryset, view):
```
[the remark this answers is not in the document]
Comment (ukanga) [reply onadata/libs/filters.py]: There is also `text`, `docs`, ..etc. There probably is a list somewhere.

File: onadata/libs/filters.py [hunk 3]
```diff
@@ -399,6 +399,19 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class ImageAttachmentFilter(filters.BaseFilterBackend):
```
Comment (ukanga) [inline onadata/libs/filters.py]: Could this class perhaps be renamed to `AttachmentTypeFilter`? It more fits what the filtering is on.

File: onadata/libs/filters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (WinnyTroy) [reply onadata/libs/filters.py]: > We might need documentation for this.

@lincmba @ukanga Could the documentation for this come in another PR?

File: onadata/libs/filters.py [hunk 4]
```diff
@@ -399,6 +399,19 @@ def filter_queryset(self, request, queryset, view):
...
+
+    def filter_queryset(self, request, queryset, view):
+        image_set = ["image/png", "image/jpeg", "image/jpg"]
```
[the remark this answers is not in the document]
Comment (WinnyTroy) [reply onadata/libs/filters.py]: sure

File: onadata/libs/filters.py (hunk 3 above)
Comment (WinnyTroy) [reply onadata/libs/filters.py]: Sounds okay. Done.

File: onadata/libs/filters.py [hunk 5]
```diff
@@ -399,6 +401,24 @@ def filter_queryset(self, request, queryset, view):
...
+        # return instances with image specific attachments.
+        if attachment_type == "image":
+            queryset = queryset.filter(mimetype__in=MEDIA_IMAGE_TYPES)
+        elif attachment_type == "audio":
+            queryset = queryset.filter(mimetype__in=MEDIA_AUDIO_TYPES)
+        elif attachment_type == "video":
+            queryset = queryset.filter(mimetype__in=MEDIA_VIDEO_TYPES)
+        elif attachment_type == "file":
+            queryset = queryset.filter(mimetype__in=MEDIA_FILE_TYPES)
+
```
Comment (ukanga) [inline onadata/libs/filters.py]: Feels like too many if statements, we could refactor to use a dictionary for all this types, i.e.
```python
MEDIA_FILE_TYPES = {
  "audio": ["audio/mp3", "audio/x-m4a", ".."],
    "documents": ["application/msword", "application/vnd.ms-excel", "..."],
    "image": ["image/jpeg", "image/png", "..."],
    ...
}
...
mime_types = MEDIA_FILE_TYPES.get(attachment_type)

if mime_types:
   queryset = queryset.filter(mimetype__in=mimetypes)
````

File: onadata/libs/utils/common_tags.py [hunk 6]
```diff
@@ -90,6 +90,13 @@
 
 API_TOKEN = 'api-token'
 KNOWN_MEDIA_TYPES = ['photo', 'image', 'audio', 'video']
+MEDIA_IMAGE_TYPES = ["image/png", "image/jpeg", "image/jpg"]
+MEDIA_VIDEO_TYPES = ["video/mp4"]
+MEDIA_AUDIO_TYPES = ["audio/mp3", "audio/mp4"]
+MEDIA_FILE_TYPES = [
+    "application/pdf", "text/plain",
```
[the remark this answers is not in the document]
Comment (WinnyTroy) [reply onadata/libs/utils/common_tags.py]: sure. Let me do that

**Pull request merged**